server/src/main/java/io/crate/session/Session.java

```diff
@@ -940,4 +980,56 @@ public String toString() {
             ", id=" + id +
             "}";
     }
+
+    /**
+     * Controls execution time of all lifecycle phases of a statement (parse/analysis/plan/execution).
+     * If statement_timeout is specified, statement is stopped when processing exceeds it.
+     */
+    public static class TimeoutToken {
+
+        private TimeValue statementTimeout;
+        private long startNanos;
+        private boolean enabled = true;
+
+        public TimeoutToken(TimeValue statementTimeout, long startNanos) {
+            this.statementTimeout = statementTimeout;
+            this.startNanos = startNanos;
+        }
+
+        /**
+         * Noop token, used for system queries and in tests.
+         */
+        public TimeoutToken() {
```

Comment:
```suggestion
        public static TimeoutToken noopToken() {
              return TimeoutToken(false);
        }
```

minor: This would be easier to use, no need to read Javadoc.

Reply:
Adjusted a bit in https://github.com/crate/crate/pull/17095/commits/3640e16df98a161775131738c0f613320f127a6d